ttnn/cpp/ttnn/graph_processor.cpp

```diff
@@ -0,0 +1,450 @@
+#include "graph_processor.hpp"
+#include "tt_metal/tt_stl/reflection.hpp"
+#include "types.hpp"
+
+
+#include <cxxabi.h>
+#include <memory>
+#include <typeindex>
+#include <unordered_map>
+
+namespace {
+const std::string demangle(const char* name) {
+
+    int status = -4;
+
+    char* res = abi::__cxa_demangle(name, NULL, NULL, &status);
+
+    const char* const demangled_name = (status==0)?res:name;
+
+    std::string ret_val(demangled_name);
+
+    free(res);
+
+    return ret_val;
+}
+
+std::string tensorMemoryLayoutToString(TensorMemoryLayout layout) {
```

Comment:
Did you want to avoid using `magic_enum` here?

Reply:
Totally forgot about it. Will add.